tests/test_agpypeline.py

```diff
@@ -0,0 +1,487 @@
+    assert orig == result
+
+
+def test_geoimage_clip_raster():
+    """Checks the dimensions when calling clip_raster"""
+    result = geoimage.clip_raster(TEST_IMAGE, (0, 0, 0, 0))
+    assert len(result) == 730 and len(result[0]) == 891
+
+
+def test_geoimage_clip_raster_intersection():
+    """Checks the output when calling clip_raster_intersection on an image, using metadata from the data folder
+    in order to make sure that the function output is correct"""
+    file_bounds = geoimage.get_image_bounds(TEST_IMAGE)
+    geo_json = json.load(open(JSON_FILE))
+    for feature in range(len(geo_json["features"])):
+        geometry = geo_json["features"][feature]["geometry"]
+        plot_bounds = ogr.CreateGeometryFromJson(str(geometry))
+        bad_result = geoimage.clip_raster_intersection(TEST_IMAGE, file_bounds, plot_bounds,
+                                                       "images/geoimage_clip_raster_intersection.tif")
+        assert bad_result is None
+    complete_overlap = str(geoimage.clip_raster_intersection(TEST_IMAGE, file_bounds, file_bounds,
+                                                             "images/geoimage_clip_raster_intersection.tif"))
+    with open("data/geoimage_clip_raster_intersection.txt", 'r') as checkfile:
+        checkfile_content = checkfile.read()
+        assert checkfile_content == complete_overlap
+
+
+def test_geoimage_clip_raster_intersection_json():
+    """Checks the output when calling clip_raster_intersection_json on an image, usingmetadata from the
+    data folder in order to make sure that the function output is correct"""
+    file_bounds = geoimage.get_image_bounds(TEST_IMAGE)
+    json_file_bounds = geometries.geometry_to_geojson(file_bounds)
+    geo_json = json.load(open(JSON_FILE))
+    for feature in range(len(geo_json["features"])):
+        json_plot_bounds = geo_json["features"][feature]["geometry"]
+        bad_result = geoimage.clip_raster_intersection_json(TEST_IMAGE, json_file_bounds, json_plot_bounds,
+                                                            "images/geoimage_clip_raster_intersection.tif")
+        assert bad_result is None
+    complete_overlap = geoimage.clip_raster_intersection_json(TEST_IMAGE, json_file_bounds, json_file_bounds,
+                                                              "images/geoimage_clip_raster_intersection.tif")
+    with open("data/geoimage_clip_raster_intersection.txt", 'r') as checkfile:
+        checkfile_content = checkfile.read()
+        assert checkfile_content == str(complete_overlap)
+
+
+def test_geoimage_create_geotiff():
+    """Tests create_geotiff, although a complete image is not generated at the moment. gps_bounds were extracted
+    from output.tin.tif"""
+    gps_bounds = (3971937.000, 3972667.000, 368908.000, 369799.000)
+    data = gdal.Open(TEST_IMAGE)
+    myarray = np.array(data.GetRasterBand(1).ReadAsArray())
+    if os.path.isfile("images/output.tif"):
+        os.remove("images/output.tif")
```

Comment:
Nice!

Reply:
This one didn't seem to change the image; I used the data from gdalinfo on the original image but it looks like only a small band was created